src/utilities/getIntrospectionQuery.ts

```diff
@@ -254,25 +254,25 @@ export type IntrospectionNonNullTypeRef<
 
 export type IntrospectionTypeRef =
   | IntrospectionNamedTypeRef
-  | IntrospectionListTypeRef
+  | IntrospectionListTypeRef<any>
```

Comment:
TS should support circular refs after 3.7, right?

Reply:
Filed an issue https://github.com/microsoft/TypeScript/issues/42308 and also opened a question on [StackOverflow](https://stackoverflow.com/questions/65693844/using-recursive-type-aliases-gives-circular-reference-error) for help.